### PR TITLE
throw an exception if arg_max or arg_min have fewer than 2 arguments

### PR DESCRIPTION
**Pull request by bemitc**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### CODE REVIEW

File: src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp [hunk 1]
```diff
@@ -23,44 +24,58 @@ namespace DB
...
         ++pos;
+        ++arg_count;
         const auto expr_to_return = getConvertedArgument(fn_name, pos);
         if (expr_to_return == expr_to_maximize)
         {
             continue;
         }
         out += std::format("argMax({}, {}) as {},", expr_to_return, expr_to_maximize, expr_to_return);
     }
+    if (arg_count < 2)
```
Comment (kashwy) [inline src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp:46]: I think Laszlo has an implementation for validating the number of args, @ltrk2

Comment (ltrk2) [reply src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp:46]: @bemitc you can use `IParserKQLFunction::getArguments`.

File: src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp [hunk 2]
```diff
@@ -27,19 +28,19 @@ bool ArgMax::convertImpl(String & out, IParser::Pos & pos)
...
     if (fn_name.empty())
         return false;
-    ++pos;
-    String expr_to_maximize = getConvertedArgument(fn_name, pos);
-    while (pos->type == TokenType::Comma)
+
+    const Interval & argument_count_interval = {2, Interval::max_bound};
+    const auto args = getArguments(fn_name, pos, ArgumentState::Parsed, argument_count_interval);
+    
+    for (std::string expr_to_return : args | std::views::drop(1))
```
Comment (ltrk2) [inline src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp:35]: You could just go with:
```suggestion
    for (const auto& expr_to_return : args | std::views::drop(1))
```

File: src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp [hunk 3]
```diff
@@ -27,19 +28,19 @@ bool ArgMax::convertImpl(String & out, IParser::Pos & pos)
...
     {
-        ++pos;
-        const auto expr_to_return = getConvertedArgument(fn_name, pos);
-        if (expr_to_return == expr_to_maximize)
+        if (expr_to_return != args[0])
```
Comment (ltrk2) [inline src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp:37]: You could just affix this to `args | std::views::drop(1)` as `std::views::filter(..)`

File: src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp [hunk 4]
```diff
@@ -49,19 +50,18 @@ bool ArgMin::convertImpl(String & out, IParser::Pos & pos)
...
         {
-            continue;
+            out += std::format("argMin({}, {}) as {},", expr_to_return, args[0], expr_to_return);
         }
```
Comment (ltrk2) [inline src/Parsers/Kusto/KustoFunctions/KQLAggregationFunctions.cpp:61]: I'd seek to apply the same modifications as with `arg_max`.

**Review by ltrk2 — CHANGES_REQUESTED** [review]

**Review by ltrk2 — APPROVED** [review]

**Pull request merged**